python/lsst/daf/butler/core/datastore.py

```diff
@@ -92,7 +92,7 @@ def registerUndo(self, name, undoFunc, *args, **kwargs):
         self._log.append(self.Event(name, undoFunc, args, kwargs))
 
     @contextlib.contextmanager
-    def undoWith(self, name, undoFunc, *args, **kwargs):
+    def undoWith(self, name: str, undoFunc: Callable, *args: Any, **kwargs: Any) -> Generator:
```

Comment:
Hmm.  I think this is technically correct, but I've been using the return type of the decorated function (i.e. `typing.ContextManager`) in this case, because I find it more useful for human readers of the code.  But I suppose if we ever did get to the point of running mypy, it'd have to be like this, i.e. the return type of the pre-decorated function?